### DISCUSSION TITLE
How I can add velocitycommand to a ros2 control node

### DISCUSSION THREAD

**Discussion by JiaheXu** [bdaiinstitute/spot_ros2#378] [Q&A] [answered]
Hi BDAI,

I am following the spot_examples/walk_forward.py. I am trying to add a new callback function in ros2 for synchro_velocity_command(). For example, send a twist msg to the ros2 node and the ros2 node sends a synchro_velocity_command() for the spot. How can I add a subscriber for bdai_ros2_wrappers?

**Comment by khughes-bdai** [comment, marked as answer]
Hi @JiaheXu ,
The driver already has a subscription to the `cmd_vel` topic. This internally calls synchro velocity commands: https://github.com/bdaiinstitute/spot_ros2/blob/d103f6fe01681251a787fe68c1caa69d1db28671/spot_driver/spot_driver/spot_ros2.py#L2557
and
https://github.com/bdaiinstitute/spot_wrapper/blob/6548b7a899a517bc24f840ba5eb36519b109ea4c/spot_wrapper/wrapper.py#L1262
You should be able to use this to publish twists and move the robot with this, let me know if you run into issues.
